File: TCPDataStreaming/src/main/java/com/projects/bigdata/data_streaming/utility/StreamingLineFactory.java
package com.projects.bigdata.data_streaming.utility;

import java.util.HashMap;
import java.util.Map;
import java.util.function.Supplier;

import com.projects.bigdata.utility.StreamingLineType;

public final class StreamingLineFactory {
	private static final Map<StreamingLineType, Supplier<String>> streamingLineMap = new HashMap<>() {
		private static final long serialVersionUID = 1L;

	{
		put(StreamingLineType.PHRASE, StreamingLineSupplier::randomPhrase);
		put(StreamingLineType.TRADE, StreamingLineSupplier::randomTrade);
	}};
	
	public static Supplier<String> getStreamingLine(final StreamingLineType streaminLineType)  {
		return streamingLineMap.get(streaminLineType);
	}
}

File: TCPDataStreaming/src/main/kotlin/com/projects/bigdata/data_streaming/utility/StreamingLineFactory.kt
package com.projects.bigdata.data_streaming.utility

import com.projects.bigdata.utility.StreamingLineType

import java.util.HashMap
import java.util.function.Supplier

object StreamingLineFactory {
    private val streamingLineMap = mapOf<StreamingLineType, () -> String>(StreamingLineType.PHRASE to StreamingLineSupplier::randomPhrase,
            StreamingLineType.TRADE to StreamingLineSupplier::randomTrade)

    fun getStreamingLine(streamingLineType: StreamingLineType): () -> String {
        return streamingLineMap[streamingLineType]!!
    }
}
